test/e2e/deployment.go

```diff
@@ -437,14 +437,14 @@ func testRolloverDeployment(f *framework.Framework) {
 	// Create a deployment to delete nginx pods and instead bring up redis-slave pods.
 	// We use a nonexistent image here, so that we make sure it won't finish
 	deploymentName, deploymentImageName := "test-rollover-deployment", "redis-slave"
-	deploymentReplicas := int32(3)
+	deploymentReplicas := int32(1)
 	deploymentImage := "gcr.io/google_samples/gb-redisslave:nonexistent"
 	deploymentStrategyType := extensions.RollingUpdateDeploymentStrategyType
 	framework.Logf("Creating deployment %q", deploymentName)
 	newDeployment := framework.NewDeployment(deploymentName, deploymentReplicas, deploymentPodLabels, deploymentImageName, deploymentImage, deploymentStrategyType)
 	newDeployment.Spec.Strategy.RollingUpdate = &extensions.RollingUpdateDeployment{
-		MaxUnavailable: func(i int) *intstr.IntOrString { x := intstr.FromInt(i); return &x }(1),
-		MaxSurge:       func(i int) *intstr.IntOrString { x := intstr.FromInt(i); return &x }(1),
+		MaxUnavailable: intOrStrP(0),
```

Comment:
Previously the deployment was using 3 replicas and a maxUnavailable of 1. This meant that rollover could be exercised, eg.

rs-1: 3 pods
==update==>
rs-1: 2 pods, rs-2: 2 pods
==deployment blocks because rs-2 is broken==>
==new update==>
rs-1 and rs-2 are rolled over by rs-3

Now that the deployment uses 1 pod, we need maxUnavailable=0 for continuing to exercise rollover